handshake_docs/handshake_flow.md

```diff
@@ -0,0 +1,35 @@
+# Handshake Flow for DIDComm Mediator-rs
+
+## 1. Communication Initiation
+### Participant Identification:
+- Each mobile agent generates a DID (Decentralized Identifier).
+
+- Agents exchange their DIDs to identify each other. The initial exchange might involve sending a DIDComm message that includes the agent’s DID and a nonce (a random value used for security purposes).
+### Initial Message:
+- The initiating agent sends a DIDComm handshake message to the mediator. This message contains:
+
+  - **DID of the initiating agent**: Identifies the sender.
+  - **Nonce**: Ensures message freshness and prevents replay attacks.
+  - **Service Information**: May include endpoint URLs, service types, and other metadata relevant to the communication.
+## 2. Message Exchanges
+### Encryption Methods:
+ - Messages are encrypted using DIDComm’s encryption standards, typically using asymmetric encryption (e.g., using public keys derived from the DIDs).
+### Data Formats:
+ - Messages are structured in JSON format, following the DIDComm v2 specification. A typical message contains:
+   - **Headers**: Metadata about the message (e.g., type, timestamp, sender, recipient).
+   - **Payload**: The actual content of the message, encrypted as needed.
+   - **Encryption Details**: Information about the encryption used, such as keys and algorithms.
+### Sequence of Exchanges:
+ - **Mediator Acknowledgment**: The mediator responds with its own DID and a nonce, acknowledging the receipt of the initial message.
+   - **Capability Negotiation**: Both parties may exchange additional messages to negotiate capabilities, such as supported encryption algorithms, message formats, and routing mechanisms.
+   - **Routing Information**: The mediator may provide routing information necessary for forwarding future messages between the agents.
+   - **Verification**: Each party verifies the identity of the other using cryptographic proofs, ensuring the authenticity and integrity of the exchanged DIDs and messages.
+## 3. Finalization
+### Successful Connection:
+ - The handshake is finalized when both parties have successfully exchanged all necessary cryptographic information and have verified each other’s identities.
```

Comment:
I am not sure of the identity verification. If most of the cases we will have no trust authority. So identities are self claimed.  This is why i would leave out the identity verification.

An out of band exchange of the initial contact is trust enough to start.